reroll-initiative/reroll-initiative.js

```diff
@@ -0,0 +1,205 @@
+/**
+ * @name Reroll-Initiative
+ * @version 0.2
+ * @author Evan Clarke <errational>
+ * @description Rerolls initiative on combat round change
+ */
+
+ /**
+  * --------------------
+  * Set module constants
+  * --------------------
+  */
+ const RRI_DEFAULT_SETTINGS = {
+    reroll: true,
+    actorTypes: "all"
+}
+  
+
+/**
+ * @class RerollInitiative
+ * @description Hooks on combat update and rerolls initiative for all combatants
+ * @todo Add configurability for when to reroll and for whom, make enable setting define whether the hook is registered or not
+ */
+class RerollInitiative {
+    settings = {};
+
+    constructor() {
+        this.postUpdateCombatHook();
+        //this.settings = {};
+        this._registerSettings();
+    }
+
+    /**
+     * module settings
+     * @todo need to store settings as an object so they can easily be retrieved
+     */
+    _registerSettings () {
+        game.settings.register("reroll-initiative", "rriSettings", {
+            name: "Reroll-Initiative Settings",
+            hint: "Settings for Reroll-Initiative module",
+            default: RRI_DEFAULT_SETTINGS,
+            type: Object,
+            scope: "world",
+            onChange: setting => {
+                console.log("settings changed, new values: ",setting)
+                //this.settings = JSON.stringify(setting);
+            }
+        });
+
+        this._loadSettings();
+
+        /* add settings as object instead
+        game.settings.register('reroll-initiative', "rriStatus", {
+            name: "Reroll-Initiative Status",
+            hint: "Enable the rerolling initiative true/false",
+            default: RRI_CONFIG.reroll,
+            //default: "",
+            type: Boolean,
+            scope: "world",
+            onChange: setting => {
+              console.log("setting change",setting);
+            }
+        });
+
+        game.settings.register('reroll-initiative', "actorTypesToReroll", {
+            name: "Actor Types to Reroll",
+            hint: "Specify the actor types to reroll PCs/NPCs/All",
+            default: RRI_CONFIG.actorTypes,
+            //default: "",
+            type: String,
+            scope: "world",
+            onChange: setting => {
+              console.log("setting change",setting);
+            }
+        });
+        */
+    }
+
+    /**
+     * Returns the default module settings
+     */
+    _defaultSettings() {
+        this.settings = RRI_SETTINGS;
+        console.log("Restting reroll-initiative settings to defaults:",RRI_DEFAULT_SETTINGS);
+    }
+
+    /**
+     * Saves current class instance settings back to game settings
+     */
+    _saveSettings () {
```

Comment:
I'd expect this method to have the settings passed to it rather than relaying on the local variable. 

It will save guard against timing issues that can happen if someone misuses this function

Reply:
maybe my intent isn't clear with this method: this is purely for use within the class to save back the contents of settings to the global game settings